test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,212 @@
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	command = []string{
+		"bash", "-c",
+		"cp nft/firewall.nft /host/etc/nftables/firewall.nft",
+	}
+
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to move NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	err = editNftablesConf(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	command = []string{
+		"bash", "-c",
+		"/usr/sbin/nft list ruleset > /etc/nftables.conf",
+	}
+
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	command = []string{"bash", "-c",
+		"systemctl enable nftables",
+	}
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to enable nftables %q on node %s: %w", command, nodeName, err)
+	}
+
+	err = saveNFTablesRules("commatrix", "nftables-"+nodeName, string(output))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func RulesList(cs *client.ClientSet, node v1.Node) ([]byte, error) {
+	utilsHelpers := utils.New(cs)
+	role, err := types.GetNodeRole(&node)
+	if err != nil {
+		return nil, err
+	}
+
+	ns := "check-firewall-" + role
+	err = utilsHelpers.CreateNamespace(ns)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err := utilsHelpers.DeleteNamespace(ns); err != nil {
+			fmt.Printf("failed cleaning namespace %s: %v", ns, err)
+		}
+	}()
+
+	debugPod, err := utilsHelpers.CreatePodOnNode(node.Name, ns, consts.DefaultDebugPodImage)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create debug pod on node %s: %w", node.Name, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	err = editNftablesConf(debugPod, utilsHelpers)
+	if err != nil {
+		return nil, err
+	}
+
+	err = installNFT(debugPod, utilsHelpers)
+	if err != nil {
+		return nil, err
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return nil, err
+	}
+
+	err = saveNFTablesRules("commatrix-after-reboot", "nftables-"+node.Name, string(output))
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func editNftablesConf(debugPod *v1.Pod, u utils.UtilsInterface) error {
+	// Command to add the include statement to nftables.conf
+	command := []string{
+		"sh", "-c", `echo 'include "/etc/nftables/firewall.nft"' >> /host/etc/sysconfig/nftables.conf`,
+	}
+	_, err := u.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to edit nftables.conf on debug pod: %w", err)
+	}
+	return nil
+}
+
+func installNFT(debugPod *v1.Pod, u utils.UtilsInterface) error {
+	installCmd := []string{"bash", "-c", "yum update -y && yum install -y nftables"}
```

Comment:
we should not install stuff instide the container this will not work on disconnected env.
you can access the host and run the nftables commands from the host (chroot /host nftables)